src/Neo/Cryptography/Crypto.cs

```diff
@@ -94,24 +95,43 @@ public static bool VerifySignature(ReadOnlySpan<byte> message, ReadOnlySpan<byte
             }
             else
             {
-                ECCurve curve =
-                    pubkey.Curve == ECC.ECCurve.Secp256r1 ? ECCurve.NamedCurves.nistP256 :
-                    pubkey.Curve == ECC.ECCurve.Secp256k1 ? ECCurve.CreateFromFriendlyName("secP256k1") :
-                    throw new NotSupportedException();
-                byte[] buffer = pubkey.EncodePoint(false);
-                using var ecdsa = ECDsa.Create(new ECParameters
-                {
-                    Curve = curve,
-                    Q = new ECPoint
-                    {
-                        X = buffer[1..33],
-                        Y = buffer[33..]
-                    }
-                });
+                var ecdsa = CreateECDsa(pubkey);
                 return ecdsa.VerifyData(message, signature, HashAlgorithmName.SHA256);
             }
         }
 
+        private static readonly Dictionary<ECC.ECPoint, ECDsa> CacheECDsa = new();
+
+        /// <summary>
+        /// Create and cache ECDsa objects
+        /// </summary>
+        /// <param name="pubkey"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public static ECDsa CreateECDsa(ECC.ECPoint pubkey)
+        {
+            if (CacheECDsa.TryGetValue(pubkey, out var cache))
+            {
+                return cache;
+            }
+            ECCurve curve =
+                pubkey.Curve == ECC.ECCurve.Secp256r1 ? ECCurve.NamedCurves.nistP256 :
+                pubkey.Curve == ECC.ECCurve.Secp256k1 ? ECCurve.CreateFromFriendlyName("secP256k1") :
+                throw new NotSupportedException();
+            byte[] buffer = pubkey.EncodePoint(false);
+            var ecdsa = ECDsa.Create(new ECParameters
+            {
+                Curve = curve,
+                Q = new ECPoint
+                {
+                    X = buffer[1..33],
+                    Y = buffer[33..]
+                }
+            });
+            CacheECDsa[pubkey] = ecdsa;
```

Comment:
So it will grow indefinitely?

Reply:
> So it will grow indefinitely?

We can set a limit to it, 10000, or 20000. But now the test result is OK.

Reply:
It's easy to limit, we don't want to leak memory here, otherwise it'd be trivial to knock an individual node off the network.

Reply:
We need to limit the number of entries

Reply:
A cache should not always grow